src/pyhf/tensor/numpy_backend.py

```diff
@@ -2,17 +2,21 @@
 from __future__ import annotations
 
 import logging
-from typing import Callable, Generic, Mapping, Sequence, TypeVar, Union
+from typing import TYPE_CHECKING, Callable, Generic, Mapping, Sequence, TypeVar, Union
 
 import numpy as np
-from numpy.typing import ArrayLike, DTypeLike, NBitBase, NDArray
+
+# Needed while numpy lower bound is older than v1.21.0
+if TYPE_CHECKING:
+    from numpy.typing import ArrayLike, DTypeLike, NBitBase, NDArray
+
```

Comment:
should add

```else:
    NBitBase = "NBitBase"
```

Reply:
@kratsg sorry I'm still bad at typing. You want this added to the `TYPE_CHECKING` guard area? What does this do?